### PR TITLE
Add support for google_compute_region_target_tcp_proxy gcp resource

### PR DESCRIPTION
**Pull request by meerkat-b**
### Description of your changes

This change adds support of the [`google_compute_region_target_tcp_proxy`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_region_target_tcp_proxy) resource to the gcp provider.

Fixes https://github.com/upbound/provider-gcp/issues/433

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
Tested locally by applying the crds, controllers, and examples to local kind cluster:
image

- [x] Requesting the aid of contributors to help me trigger the uptest CI workflow with the command:
`/test-examples="examples/compute/regiontargettcpproxy.yaml"`

Thanks!

### CODE REVIEW

**Comment by meerkat-b** [comment]
/test-examples="provider-aws/examples/compute/regiontargettcpproxy.yaml"

**Comment by jeanduplessis** [comment]
/test-examples="provider-aws/examples/compute/regiontargettcpproxy.yaml"

**Comment by meerkat-b** [comment]
/test-examples="provider-aws/examples/compute/regiontargettcpproxy.yaml"

**Comment by jeanduplessis** [comment]
/test-examples="provider-aws/examples/compute/regiontargettcpproxy.yaml"

**Comment by jeanduplessis** [comment]
/test-examples="examples/compute/regiontargettcpproxy.yaml"

**Comment by jeanduplessis** [comment]
/test-examples="examples/compute/regiontargettcpproxy.yaml"

**Review by turkenf — COMMENTED** [review]
Thanks a lot for this PR @meerkat-b, I left two small comments for you to consider.

File: config/compute/config.go [hunk 1]
```diff
@@ -360,6 +360,16 @@ func Configure(p *config.Provider) { //nolint: gocyclo
 
 	})
 
+	p.AddResourceConfigurator("google_compute_region_target_tcp_proxy", func(r *config.Resource) {
+		config.MarkAsRequired(r.TerraformResource, "region")
+
+		r.References["backend_service"] = config.Reference{
+			Type:      "RegionBackendService",
```
[the remark this answers is not in the document]
Comment (meerkat-b) [reply config/compute/config.go:367]: Thanks for the suggestion, I'll apply that.

File: config/externalnamenottested.go [hunk 2]
```diff
@@ -309,6 +309,8 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
 	"google_compute_snapshot": config.TemplatedStringAsIdentifier("name", "projects/{{ .setup.configuration.project }}/global/snapshots/{{ .external_name }}"),
 	// Imported by using the following projects/{{project}}/global/sslPolicies/{{name}}
 	"google_compute_ssl_policy": config.TemplatedStringAsIdentifier("name", "projects/{{ .setup.configuration.project }}/global/sslPolicies/{{ .external_name }}"),
+	// Imported by using the following format: projects/{{project}}/regions/{{region}}/targetTcpProxies/{{name}}
+	"google_compute_region_target_tcp_proxy": config.TemplatedStringAsIdentifier("name", "projects/{{ .setup.configuration.project }}/regions/{{ .parameters.region }}/targetTcpProxies/{{ .external_name }}"),
```
[the remark this answers is not in the document]
Comment (meerkat-b) [reply config/externalnamenottested.go:313]: An oversight on my part, thanks.

**Comment by meerkat-b** [comment]
@turkenf thanks for your comments.
I've applied the two suggestions, rebased against main, and also re-tested after changes against my local kind cluster.

image

**Comment by turkenf** [comment]
/test-examples="examples/compute/regiontargettcpproxy.yaml"

**Review by turkenf — APPROVED** [review]
Thanks for your nice work @meerkat-b, LGTM.

**Comment by meerkat-b** [comment]
thanks for the prompt review!

**Pull request merged**